test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -315,6 +315,42 @@ def __repr__(self):
                 str(self.rv) + ', msg=' + str(self.msg) + ')')
 
 
+class PtlfailureException(AssertionError):
+
+    """
+    Generic failure exception raised by PTL operations.
+    Sets a ``return value``, a ``return code``, and a ``message``
+    A post function and associated positional and named arguments
+    are available to perform any necessary cleanup.
+
+    :param rv: Return value set for the failure occured during PTL
+               operation
+    :type rv: int or None.
+    :param rc: Return code set for the failure occured during PTL
+               operation
+    :type rc: int or None.
+    :param msg: Message set for the failure occured during PTL operation
+    :type msg: str or None.
+    :param post: Execute necessary cleanup if not None
```

Comment:
Yes, modified doc for same. But regarding prototype of callback function, its free form, there is no defined prototype for it. You can choose any number of argument with any type of argument to that callback function and which I don't think we need to specify that here.